test/FBeetsRevenueSharer.test.ts

```diff
@@ -123,7 +125,84 @@ describe("BeetsBar", function () {
+  })
+
+  it("distribute correct amount of to lockers after changing share", async () => {
+    const sharer: FBeetsRevenueSharer = await deployFBeetsSharer()
+    await chef.add(10, sharer.address, ethers.constants.AddressZero)
+
+    // we need to add the sharer as a rewarder to the locker
+    await locker.addReward(beets.address, sharer.address)
+    // change the default share
+    await sharer.setFBeetsLockerShare(750)
+    // also we need someone to lock some fBeets
+    const lockAmount = bn(1, 0)
+    await mintFBeets(bob, lockAmount)
+    await beetsBar.connect(bob).approve(locker.address, lockAmount)
+    await locker.connect(bob).lock(bob.address, lockAmount)
+
+    await sharer.depositToChef()
+    // lets advance some blocks to generate some emissions
+    await advanceBlocks(10)
+    await sharer.harvestAndDistribute()
+
+    // now lets see if the locker has 75% of the claimed rewards
+    const userInfo = await chef.userInfo(0, sharer.address)
+    expect(await beets.balanceOf(locker.address)).to.equal(userInfo.rewardDebt.mul(await sharer.fBeetsLockerShare()).div(await sharer.DENOMINATOR()))
+    // to go full circle, we advance the full reward duration which is 1 epoch
+    await advanceTime(EPOCH_DURATION)
+    // // now lets claim the rewards
+    await locker.getReward(bob.address)
+    // //  bob should have now all the rewards on the locker or 75% of the emissions (minus some rounding errors)
+    expect(await beets.balanceOf(bob.address)).to.be.closeTo(userInfo.rewardDebt.mul(await sharer.fBeetsLockerShare()).div(await sharer.DENOMINATOR()), 200000)
+  })
+
+  it("distribute 100% to lockers and none to non-lockers", async () => {
+    const sharer: FBeetsRevenueSharer = await deployFBeetsSharer()
+    await chef.add(10, sharer.address, ethers.constants.AddressZero)
+
+    // we need to add the sharer as a rewarder to the locker
+    await locker.addReward(beets.address, sharer.address)
+    // change the default share
+    await sharer.setFBeetsLockerShare(1000)
+    // also we need someone to lock some fBeets
+    const lockAmount = bn(1, 0)
+    await mintFBeets(bob, lockAmount)
+    await beetsBar.connect(bob).approve(locker.address, lockAmount)
+    await locker.connect(bob).lock(bob.address, lockAmount)
+    // also we need someone to NOT lock fBeets, just mint
+    await mintFBeets(alice, lockAmount)
+
+    await sharer.depositToChef()
+    // lets advance some blocks to generate some emissions
+    await advanceBlocks(10)
+    await sharer.harvestAndDistribute()
+
+    // now lets see if the locker has 100% of the rewards
+    const userInfoBob = await chef.userInfo(0, sharer.address)
+    expect(await beets.balanceOf(locker.address)).to.equal(userInfoBob.rewardDebt.mul(await sharer.fBeetsLockerShare()).div(await sharer.DENOMINATOR()))
+    // now lets see if alice has zero rewards
+    const userInfoAlice = await chef.userInfo(1, sharer.address)
```

Comment:
there is no farm with id `1`, but since everything is initialized with default values (there is no null), the rewardDebt is `0`, since its the default value. so this makes the next expect test pass by accident